build.sh

```diff
@@ -1,105 +1,142 @@
-#!/bin/bash
-# Build lexical models
-
-# Set sensible script defaults:
-# set -e: Terminate script if a command returns an error
-set -e
-# set -u: Terminate script if an unset variable is used
-set -u
-
+#!/usr/bin/env bash
 #
 # This script is built with commands available to Git Bash on Windows. (mingw32)
-#
 
-function display_usage {
-  echo "Usage: $0 [-t(est)|-b(uild)|-c(lean)] [-no-npm] [-s] [-color|-no-color] [target]"
-  echo "  -t || -test   Runs tests on models"
-  echo "  -b || -build  Creates compiled models"
-  echo "  -c || -clean  Cleans intermediate and output files"
-  echo "  -color        Force color output"
-  echo "  -no-color     Force no color output"
-  echo "  -no-npm       Skip all npm steps"
-  echo "  -s            Quiet build"
-  echo "  target        The specific model(s) to build, e.g. release or release/example/en.template"
-  echo "                If omitted, builds all models"
-  exit 1
-}
+# Prevents 'clear' on exit of mingw64 bash shell
+SHLVL=0
 
-#
-# Define paths
-#
-MODELROOT="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+## START STANDARD BUILD SCRIPT INCLUDE
+# adjust relative paths as necessary
+THIS_SCRIPT="$(readlink -f "${BASH_SOURCE[0]}")"
+. "${THIS_SCRIPT%/*}/resources/builder.inc.sh"
+# END STANDARD BUILD SCRIPT INCLUDE
 
-# Master json schema is from https://api.keyman.com/schemas/model_info.json
+# script runs from repo root
+cd "$THIS_SCRIPT_PATH"
 
-MODELINFO_SCHEMA_JSON="$MODELROOT/tools/model_info.source.json"
-MODELINFO_SCHEMA_DIST_JSON="$MODELROOT/tools/model_info.distribution.json"
+. "./resources/util.inc.sh"
+. "./resources/build_targets.inc.sh"
+. "./resources/external.inc.sh"
 
-. "$MODELROOT/resources/util.sh"
-. "$MODELROOT/resources/compile.sh"
-. "$MODELROOT/resources/external.sh"
-#. "$MODELROOT/resources/validate.sh"
-#. "$MODELROOT/resources/merge.sh"
+builder_describe \
+  "Build Keyman lexical models" \
+  clean \
+  configure \
+  build \
+  test \
+  "--model,-m=MODEL          Build specific targets only, e.g. release/ or release/a/ or release/a/anii/, comma separated" \
+  "--silent,-s               Suppress unnecessary messages"
```

Comment:
TODO to update https://github.com/keymanapp/lexical-models/blob/master/README.md#build-instructions